### PR TITLE
EES-3257 include non numerical data option for charts

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Adds an option to Chart configuration to include data sets with non-numerical values. When this is checked the data set is shown with no bar/line.

include

### CODE REVIEW

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/GovUk.Education.ExploreEducationStatistics.Common/Model/Chart/Charts.cs [hunk 1]
```diff
@@ -26,6 +27,7 @@ public abstract class Chart : IChart
         public string Alt { get; set; }
         public int Height { get; set; }
         public int? Width { get; set; }
+        public bool? IncludeNonNumericalData { get; set; }
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Common/Model/Chart/Charts.cs]: ok, done.

File: src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartAxisConfiguration.tsx [hunk 2]
```diff
@@ -49,6 +49,7 @@ interface Props {
   definition: ChartDefinition;
   data: TableDataResult[];
   meta: FullTableMeta;
+  includeNonNumericalData: boolean;
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartAxisConfiguration.tsx]: ok, done.

File: src/explore-education-statistics-common/src/modules/charts/types/chart.ts [hunk 3]
```diff
@@ -76,6 +76,7 @@ export interface ChartProps {
   width?: number;
   axes: AxesConfiguration;
   legend?: LegendConfiguration;
+  includeNonNumericalData: boolean;
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-common/src/modules/charts/types/chart.ts]: ok, done.

File: src/explore-education-statistics-common/src/modules/charts/types/chart.ts [hunk 4]
```diff
@@ -104,6 +105,7 @@ export interface ChartDefinitionOptions {
   title?: string;
   titleType: 'default' | 'alternative';
   alt: string;
+  includeNonNumericalData: boolean;
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-common/src/modules/charts/types/chart.ts]: It's needed here for the chart config form.

File: src/explore-education-statistics-common/src/modules/charts/util/createDataSetCategories.ts [hunk 5]
```diff
@@ -321,18 +322,29 @@ export default function createDataSetCategories(
...
 
-      const value = Number(get(measuresByDataSet, getIndicatorPath(dataSet)));
+    if (!Number.isNaN(value)) {
+      return [...acc, [childDataSet, value] as Pair<ChildDataSet, number>];
+    }
 
-      return [childDataSet, value] as Pair<ChildDataSet, number>;
-    })
-    .filter(([, value]) => !Number.isNaN(value));
+    return acc;
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-common/src/modules/charts/util/createDataSetCategories.ts]: yeah good point. Sorted.

**Review by ntsim — APPROVED** [review]

**Pull request merged**